pkg/utils/pod.go

```diff
@@ -48,13 +48,26 @@ func IsPodReadyConditionTrue(status v1.PodStatus) bool {
 	return condition != nil && condition.Status == v1.ConditionTrue
 }
 
+// IsContainerReadyConditionTrue returns true if a containers are ready; false otherwise.
+func IsContainerReadyConditionTrue(status v1.PodStatus) bool {
+	condition := GetContainersReadyCondition(status)
+	return condition != nil && condition.Status == v1.ConditionTrue
+}
+
 // GetPodReadyCondition extracts the pod ready condition from the given status and returns that.
 // Returns nil if the condition is not present.
 func GetPodReadyCondition(status v1.PodStatus) *v1.PodCondition {
 	_, condition := GetPodCondition(&status, v1.PodReady)
 	return condition
 }
 
+// GetContainersReadyCondition extracts the containers ready condition from the given status and returns that.
+// Returns nil if the condition is not present.
+func GetContainersReadyCondition(status v1.PodStatus) *v1.PodCondition {
```

Comment:
Shouldn't we check all the containers ready?

Reply:
it check the pod condition, still pod level. I notice it's probably my mistake and I forgot double check the target-pod. https://github.com/vllm-project/aibrix/issues/781 so we do not have code issue and I think we do not need this PR. Pod will be ready only after all container become ready. I was talking out `running` but not `ready` in original issue. 

```
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2025-03-04T03:48:14Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2025-03-04T03:51:56Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2025-03-04T03:51:56Z"
    status: "True"
    type: ContainersReady
```